socket_windows.go

```diff
@@ -0,0 +1,181 @@
+package kr
```

Comment:
It looks like most of these functions didn't change from `socket.go`. Can we split these into
`socket.go` for common functions
`socket_windows.go` for windows-specific impl (looks like `KrDir()` and a way to `Listen` or `Dial` should be the only windows specific ones)
`socket_unix.go` for linux/darwin/freebsd impls of `KrDir`, `Listen`, `Dial`